iterator_test.go

```diff
@@ -2722,6 +2724,182 @@ func BenchmarkSeekPrefixTombstones(b *testing.B) {
 	}
 }
 
+func waitForCompactions(d *DB) {
+	d.mu.Lock()
+	// NB: Wait for table stats because some compaction types rely
+	// on table stats to be collected.
+	d.waitTableStats()
+	for d.mu.compact.compactingCount > 0 {
+		d.mu.compact.cond.Wait()
+		d.waitTableStats()
+	}
+	d.mu.Unlock()
+}
+
+// BenchmarkPointDeletedSwath benchmarks iterator operations on large-ish
+// (hundreds of MBs) databases containing broad swaths of keys removed by point
+// tombstones.
+func BenchmarkPointDeletedSwath(b *testing.B) {
+	const maxKeyLen = 5
+	ks := testkeys.Alpha(maxKeyLen)
+
+	opts := func() *Options {
+		return (&Options{
+			DisableWAL:         true,
+			FS:                 vfs.NewMem(),
+			Comparer:           testkeys.Comparer,
+			FormatMajorVersion: FormatNewest,
+		}).EnsureDefaults()
+	}
+	type iteratorOp struct {
+		name string
+		fn   func(*DB, testkeys.Keyspace, *rand.Rand) error
+	}
+	var iterKeyBuf [maxKeyLen]byte
+
+	iterOps := []iteratorOp{
+		{
+			name: "prefix point lookup", fn: func(d *DB, ks testkeys.Keyspace, rng *rand.Rand) error {
+				n := testkeys.WriteKey(iterKeyBuf[:], ks, int64(rng.Intn(int(ks.Count()))))
+				iter, _ := d.NewIter(nil)
+				_ = iter.SeekPrefixGE(iterKeyBuf[:n])
```

Comment:
One issue with the way these operations are written, is that the dominant thing is going to be iter opening/closing (which grabs version locks, and grabs an object from the pool, etc.) and not the actual operation you're trying to benchmark. Try reusing the iterator and just doing `SeekPrefixGE` etc. `b.N` times?